asmcomp/linearize.ml

```diff
@@ -120,18 +121,49 @@ let check_label n = match n.desc with
   | Llabel lbl -> lbl
   | _ -> -1
 
+
+(* Add pseudo-instruction Ladjust_trap_depth in front of a continuation
+   to notify assembler generation about updates to the stack as a result
+   of differences in exception trap depths.
+   The argument delta is the number of trap frames (not bytes). *)
+
+let adjust_trap_depth delta_traps next =
+  (* Simplify by merging and eliminating Ladjust_trap_depth instructions
+     whenever possible. *)
+  let delta_traps, next  =
+    match next.desc with
+    | Ladjust_trap_depth { delta_traps = k } -> delta_traps + k, next.next
+    | _ -> delta_traps, next
+  in
+  if delta_traps = 0 then next
+  else cons_instr (Ladjust_trap_depth { delta_traps }) next
+
```

Comment:
This code could be simplified by the magic of recursion.
```
let rec adjust_trap_depth delta_traps next =
  match next.desc with
  | Ladjust_trap_depth { delta_traps = k } ->
      adjust_trap_depth (delta_traps + k) next.next
  | _ ->
      if delta_traps = 0 then next
      else cons_instr (Ladjust_trap_depth { delta_traps }) next
```

Reply:
Thank you! Rebased and magicked.